packages/expo-permissions/ios/EXPermissions/EXLocationRequester.m

```diff
@@ -83,8 +84,11 @@ - (void)requestPermissionsWithResolver:(EXPromiseResolveBlock)resolve rejecter:(
     _resolve = resolve;
     _reject = reject;
 
-    _locMgr = [[CLLocationManager alloc] init];
-    _locMgr.delegate = self;
+    __weak typeof(self) weakSelf = self;
+    [EXUtilities performSynchronouslyOnMainThread:^{
```

Comment:
TODO: Add explanatory comment mentioning https://stackoverflow.com/a/12395192/1123156

Reply:
Also, see whether this fixed the “Location permission issue” when asking for multiple permissions in one call.

Reply:
Could you please split this out into a different commit (feel free to commit it directly to master)? This will make it easier to run git bisect or revert commits, which is generally useful and especially helpful for this massive PR.

Reply:
Done in https://github.com/expo/expo/commit/304fe560500b662be53be2c1d5a06445ad9d3702. 🙂